src/libostree/ostree-repo-commit.c

```diff
@@ -884,6 +1059,9 @@ write_metadata_object (OstreeRepo         *self,
         }
     }
 
+  if (ostree_repo_get_mode (self) == OSTREE_REPO_MODE_ARCHIVE_Z2 && self->generate_sizes)
```

Comment:
Since this check is done many times throughout the code, it would be nice for it to be in its own method.